CollAction/Frontend/app/donation/DonationBox.tsx

```diff
@@ -205,27 +230,27 @@ class DonationBox extends React.Component<IDonationBoxProps, IDonationBoxState>
             </div>);
     }
 
-    renderPopup() {
+    renderPopup(): JSX.Element {
         return (
-            <Modal isOpen={this.state.showDialog} onRequestClose={() => this.onCloseDialog()} contentLabel="Donate" style={popupStyles}>
+            <Modal isOpen={this.state.showBankDialog} onRequestClose={() => this.onCloseDialog()} contentLabel="Donate" style={popupStyles}>
                 <div id="donation-modal">
                     <StripeProvider apiKey={this.props.stripePublicKey}>
                         <Elements>
-                            <InjectedIdealBox amount={this.state.amount} userEmail={this.getEmail()} userName={this.getName()} />
+                            <InjectedDebitDetailsBox amount={this.state.amount} userEmail={this.getEmail()} userName={this.getName()} isRecurring={this.state.isRecurring} />
                         </Elements>
                     </StripeProvider>
                     <a className="btn" onClick={() => this.onCloseDialog()}>Close</a>
                 </div>
             </Modal>);
     }
 
-    renderPaymentOptions() {
+    renderPaymentOptions(): JSX.Element {
         return (<div className="payment-options">
             <div className="row">
                 <div className="col-xs-12">
-                    <button className="btn" value="ideal" onClick={() => this.payIdeal()}>
+                    <button className="btn" value="ideal" onClick={() => this.payBank()}>
                         <img src="/images/thankyoucommit/iDEAL-logo.png" />
-                        <div>iDEAL</div>
+                        <div>Debit (iDeal / SEPA Direct)</div>
```

Comment:
Does the average person know what SEPA means?

Reply:
I'm not sure. You could add it to the JIRA ticket, maybe the designers have a better idea what this button should be (maybe only debit?).